### PR TITLE
Added keypad support

### PR DESCRIPTION
**Pull request by Cybertwip**
Keypad support for onKeyPressed and onKeyReleased events when using lua bindings.

### CODE REVIEW

**Comment by pandamicro** [comment]
@dualface Please review this PR

**Comment by Cybertwip** [comment]
@pandamicro @dualface With the last commit the keypad support compiles for all platforms. It is being used a LuaValueDict to send the key and a boolean with the state of the key.

**Comment by dualface** [comment]
i agree

File: cocos/2d/CCLayer.cpp [hunk 1]
```diff
@@ -266,10 +266,21 @@ void Layer::onAcceleration(Acceleration* acc, Event* unused_event)
 #endif
 }
 
+/////////////////////////////////
+// greentwip fix for key codes //
```
Comment (pandamicro) [inline cocos/2d/CCLayer.cpp:270]: What does this comment mean ?

Comment (Cybertwip) [inline cocos/2d/CCLayer.cpp:270]: That's our organization. I did share the fork with my team and used that comment to give them proper indications about the changes done to the code.

**Comment by pandamicro** [comment]
@VictorJL Sorry that I just got time to take a look at your PR. I found that your PR provide new functionalities for deprecated functions. We don't want to encourage our user to use `setKeyboardEnabled` anymore, so I don't think this PR can be merged to our current develop branch, it may be a good addon for v2 branch though.

Instead of using `setKeyboardEnabled`, you can use `EventDispatcher::addEventListenerWithXXX` which should support keypad.

**Pull request closed without merging**